src/server.ts

```diff
@@ -245,16 +246,36 @@ function slicePayload(payload, batchSize, ecosystem): any {
     return reqData;
 }
 
+const golangExecExists = new Promise<Boolean>(resolve => {
+    exec(`which ${config.golang_executable}`, { maxBuffer: 1024 * 1200 }, error => {
+        if (error) {
+            resolve(false);
+        } else {
+            resolve(true);
+        }
+    });
+});

```

Comment:
You can use return code of exec to find the whether go is available or not.

e.g.

```
exec("invalid_cmd", (err) => console.log(err.code)); // 127
exec("go list", (err) => console.log(err.code)); // 0
exec("go list /invalid_path", (err) => console.log(err.code)); // 1
```

Use this logic to return propose error message from collector.

Reply:
This approach was good, it implemented in latest code PTAL.